kover-jvm-agent/src/main/java/kotlinx/kover/jvmagent/IntellijIntegration.java

```diff
@@ -0,0 +1,112 @@
+package kotlinx.kover.jvmagent;
+
+import com.intellij.rt.coverage.main.CoveragePremain;
+
+import java.lang.instrument.Instrumentation;
+import java.util.ArrayList;
+import java.util.List;
+
+import static kotlinx.kover.jvmagent.ParseUtils.isBoolean;
+
+public class IntellijIntegration {
+    /**
+     * A flag to enable tracking per test coverage.
+     */
+    private static final boolean TRACKING_PER_TEST = false;
+
+    /**
+     * A flag to calculate coverage for unloaded classes.
+     */
+    private static final boolean CALCULATE_FOR_UNLOADED_CLASSES = false;
+
+    /**
+     * Create hit block only for line, false adds the ability to count branches
+     */
+    private static final boolean LINING_ONLY_MODE = false;
+
+    private IntellijIntegration() {
+        // no-op
+    }
+
+    public static void callPremain(KoverAgentSettings settings, Instrumentation instrumentation) throws Exception {
+        setIntellijSystemProperties();
+        String intelliJArgsString = joinIntellijArgs(createIntellijArgs(settings));
+
+        CoveragePremain.premain(intelliJArgsString, instrumentation);
+    }
+
+    private static void setIntellijSystemProperties() {
+        System.setProperty("idea.coverage.use.system.classloader", "true");
+
+        // Disable agent logging to stdout for messages of levels `debug`, `info`, `warn`.
+        System.setProperty("idea.coverage.log.level", "error");
+
+        // Enables saving the array in the candy field,
```

Comment:
you mean cOndy field? don't remember any candies in class files :(

Reply:
🍬